### DISCUSSION TITLE
Laravel's new installation process is not compatible with Windows

### DISCUSSION THREAD

**Discussion by rinodrummer** [laravel/framework#53215] [Q&A] [answered]
Two days ago I created a new project on Windows via the Laravel installer provided by Laravel Herd (`laravel new`).

I encountered a problem during the installation:

The installation stopped since Laravel Pail requires pcntl (which is not supported on Windows), so I had to run `composer install --ignore-platform-reqs` manually to proceed.

I don't want to blame or complain, but I think that the installation process should work on every OS out of the box.

Is there some flag I have to add when running the command which isn't documented? 

Should Laravel always be installed via WSL now on?

I don't know if this topic needs the opening of an issue, so I preferred to discuss it here before.

**Comment by Sophist-UK** [comment]
Is it `Pail` itself that is requiring `pcntl` or a dependency? 

I would expect to see a line like `"ext-pcntl":"*",` in Pail's composer.json file:
```
    "require": {
        "php": "^8.2",
        "ext-mbstring": "*",
        "illuminate/console": "^10.24|^11.0",
        "illuminate/contracts": "^10.24|^11.0",
        "illuminate/log": "^10.24|^11.0",
        "illuminate/process": "^10.24|^11.0",
        "illuminate/support": "^10.24|^11.0",
        "nunomaduro/termwind": "^1.15|^2.0",
        "symfony/console": "^6.0|^7.0"
    },
```
so, I am guessing that `illuminate/process` is the culprit:
```
    "require": {
        "php": "^8.1",
        "illuminate/collections": "^10.0",
        "illuminate/contracts": "^10.0",
        "illuminate/macroable": "^10.0",
        "illuminate/support": "^10.0",
        "symfony/process": "^6.2"
    },
```
Nope, so I am guessing that `symfony/process` is the culprit:
```
    "require": {
        "php": ">=8.2"
    },
```
Nope. So no idea what is requiring `pcntl`??!!

**Reply by Sophist-UK** [reply]
A broader search shows that `illuminate/console` has:
```
    "suggest": {
        "ext-pcntl": "Required to use signal trapping.",
...
    },
```
Symfony has nothing.

**Reply by rinodrummer** [reply]
I just read it in Pail's documentation, in the notice below the [Installation chapter](https://laravel.com/docs/11.x/logging#pail-installation).

**Reply by rojtjo** [reply, marked as answer]
This was actually fixed a few days ago. Are you still experiencing this issue when using `laravel new` today?

See: https://github.com/laravel/pail/commit/b33ad8321416fe86efed7bf398f3306c47b4871b, #53172 and #53173

**Reply by Sophist-UK** [reply]
> I would expect to see a line like `"ext-pcntl":"*",` in Pail's composer.json file:
> 
> ```
>     "require": {
>         "php": "^8.2",
>         "ext-mbstring": "*",
>         "illuminate/console": "^10.24|^11.0",
>         "illuminate/contracts": "^10.24|^11.0",
>         "illuminate/log": "^10.24|^11.0",
>         "illuminate/process": "^10.24|^11.0",
>         "illuminate/support": "^10.24|^11.0",
>         "nunomaduro/termwind": "^1.15|^2.0",
>         "symfony/console": "^6.0|^7.0"
>     },
> ```

The reason I didn't see this yesterday is that Taylor removed the `"ext-pcntl": "*",` line 4 days ago.

**Reply by rinodrummer** [reply]
> This was actually fixed a few days ago. Are you still experiencing this issue when using `laravel new` today?
> 
> See: [laravel/pail@b33ad83](https://github.com/laravel/pail/commit/b33ad8321416fe86efed7bf398f3306c47b4871b), #53172 and #53173

@rojtjo, thanks for reporting it!

Yes, I tried and it works now!
